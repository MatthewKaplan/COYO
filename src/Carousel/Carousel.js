import React, { Component } from 'react';
import TypeCard from '../TypeCard/TypeCard.js';
import '../Styles/Carousel.scss';
import anusara from './ANUSARA.png'
import ashtanga from './ASHTANGA.png'
import bikram from './BIKRAM.png'
import hatha from './HATHA.png'
import iyengar from './IYENGAR.png'
import jivamukti from './JIVAMUKTI.png'
import kundalini from './KUNDALINI.png'
import vinyasa from './VINYASA.png'
import yin from './YIN.png'
import restorative from './RESTORATIVE.png'

export default class Carousel extends Component {
  constructor(props) {
    super(props);
    this.state = {
      
    }
  }

  render() {
    return (
      <section className='typeContainer'>
        <article className="imageContainer">
          <img src={anusara} alt="anusara" className="anusara typeSmall" />
          <img src={ashtanga} alt="ashtanga" className="ashtanga typeSmall" />
          <img src={bikram} alt="bikram" className="bikram typeSmall" />
          <img src={jivamukti} alt="jivamukti" className="jivamukti typeSmall" />
          <img src={iyengar} alt="iyengar" className="iyengar typeSmall" />
          <img src={hatha} alt="hatha" className="hatha typeSmall" />
          <img src={restorative} alt="restorative" className="restorative typeSmall" />
          <img src={vinyasa} alt="vinyasa" className="vinyasa typeSmall" />
          <img src={kundalini} alt="kundalini" className="kundalini typeSmall" />
          <img src={yin} alt="yin" className="yin typeSmall" />
        </article>
        {
          this.props.yogaTypes.map(yogaType =>
            <TypeCard
              name={yogaType.name}
              id={yogaType.typeId}
              purpose={yogaType.purpose}
              beginnerFriendly={yogaType.beginnerFriendly}
              propsUsed={yogaType.propsUsed}
              commonPoses={yogaType.commonPoses.map(poses => <p> {poses} </p>)}
              studios={this.props.studios}
              storeRendered={this.props.storeRendered}
            />
          )
        }
      </section>
    );
  }
}