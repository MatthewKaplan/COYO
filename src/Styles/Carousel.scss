@import 'Mixins';

.typeContainer {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  margin: 0;
  padding: 0;
  background-image: linear-gradient(-180deg, #353333
 9%,#1F6B87 100%);
}
