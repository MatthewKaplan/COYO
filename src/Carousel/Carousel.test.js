import React from 'react';
import { Carousel } from './Carousel';
import { shallow } from 'enzyme';

describe('Carousel', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Carousel data={{}} />);
    expect(wrapper).toMatchSnapshot();
  });
});