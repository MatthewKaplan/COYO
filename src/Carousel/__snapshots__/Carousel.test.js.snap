// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should match snapshot 1`] = `
<main
  className="carouselContainer"
>
  <section
    className="carouselText"
  >
    <h3
      className="carouselHeading"
    >
      Scroll to explore yoga types &gt;&gt;&gt; 
    </h3>
    <p
      className="carouselMessage"
    >
      Click a type to learn more and find a local studio offering that style
    </p>
  </section>
  <section
    className="typeContainer"
  >
    <TypeCard
      beginnerFriendly={true}
      commonPoses={
        Array [
          "Vrikshasana, ",
          "Salabhasana, ",
          "Tadasana, ",
          "Uttanasana, ",
          "Adho Mukha Svanasana, ",
          "Setu Bandhasana, ",
        ]
      }
      image="/static/media/JIVAMUKTI.41aa6479.png"
      name="Hatha"
      propsUsed={false}
      purpose="To align and calm your body, mind, and spirit in preparation for meditation."
      storeEmptyType={[MockFunction]}
      storeRendered={[MockFunction]}
      studios={
        Object {
          "address": "1735 E 17th Ave, Denver, CO 80218",
          "dropInFee": "$17.00",
          "googleRating": 4.8,
          "img": "https://i.imgur.com/w7KfQWv.jpg",
          "link": "https://wholeyoga.com",
          "mainTypeOffered": 5,
          "mission": "Where body and mind become whole.",
          "name": "Whole Yoga",
          "openEveryday": true,
          "phoneNum": "(303)333-9642",
          "teacherTraining": true,
        }
      }
      types={
        Array [
          Object {
            "beginnerFriendly": true,
            "commonPoses": Array [
              "Vrikshasana",
              "Salabhasana",
              "Tadasana",
              "Uttanasana",
              "Adho Mukha Svanasana",
              "Setu Bandhasana",
            ],
            "id": 10,
            "image": "/static/media/JIVAMUKTI.41aa6479.png",
            "name": "Hatha",
            "propsUsed": false,
            "purpose": "To align and calm your body, mind, and spirit in preparation for meditation.",
          },
        ]
      }
    />
  </section>
</main>
`;
